worlds/oot_soh/location_access/dungeons/ganons_castle.py

```diff
@@ -5,22 +5,396 @@
+        (Regions.GANONS_CASTLE_SHADOW_TRIAL, lambda bundle: True),
+        (Regions.GANONS_CASTLE_SPIRIT_TRIAL, lambda bundle: True),
+        (Regions.GANONS_CASTLE_LIGHT_TRIAL, lambda bundle: can_use(Items.GOLDEN_GAUNTLETS, bundle)),
+        (Regions.GANONS_TOWER_ENTRYWAY, lambda bundle: True),
+        (Regions.GANONS_CASTLE_DEKU_SCRUBS, lambda bundle: (can_do_trick(Tricks.LENS_GANON) or
+                                                            can_use(Items.LENS_OF_TRUTH, bundle))),
+    ])
+    ## Ganon's Castle Deku Scrubs
+    # Events
+    add_events(Regions.GANONS_CASTLE_DEKU_SCRUBS, world, [
+        (EventLocations.GANONS_CASTLE_FREE_FAIRIES, Events.CAN_ACCESS_FAIRIES, lambda bundle: True)
+    ])
+    # Locations
+    add_locations(Regions.GANONS_CASTLE_DEKU_SCRUBS, world, [
+        (Locations.GANONS_CASTLE_DEKU_SCRUB_CENTER_LEFT, lambda bundle: can_stun_deku(bundle)),
+        (Locations.GANONS_CASTLE_DEKU_SCRUB_CENTER_RIGHT, lambda bundle: can_stun_deku(bundle)),
+        (Locations.GANONS_CASTLE_DEKU_SCRUB_RIGHT, lambda bundle: can_stun_deku(bundle)),
+        (Locations.GANONS_CASTLE_DEKU_SCRUB_LEFT, lambda bundle: can_stun_deku(bundle)),
+        (Locations.GANONS_CASTLE_SCRUBS_FAIRY1, lambda bundle: True),
+        (Locations.GANONS_CASTLE_SCRUBS_FAIRY2, lambda bundle: True),
+        (Locations.GANONS_CASTLE_SCRUBS_FAIRY3, lambda bundle: True),
+        (Locations.GANONS_CASTLE_SCRUBS_FAIRY4, lambda bundle: True),
+        (Locations.GANONS_CASTLE_SCRUBS_FAIRY5, lambda bundle: True),
+        (Locations.GANONS_CASTLE_SCRUBS_FAIRY6, lambda bundle: True),
+        (Locations.GANONS_CASTLE_SCRUBS_FAIRY7, lambda bundle: True),
+        (Locations.GANONS_CASTLE_SCRUBS_FAIRY8, lambda bundle: True),
+    ])
+    ## Ganon's Castle Forest Trial
+    # Events
+    add_events(Regions.GANONS_CASTLE_FOREST_TRIAL, world, [
+        (EventLocations.GANONS_CASTLE_FOREST_TRIAL_AREA, LocalEvents.GANONS_CASTLE_FOREST_TRIAL_CLEARED, lambda bundle: (can_use(Items.LIGHT_ARROW, bundle) and
+                                                                                                                         (can_use(Items.FIRE_ARROW, bundle) or
+                                                                                                                          can_use(Items.DINS_FIRE, bundle))))
+    ])
+    # Locations
+    add_locations(Regions.GANONS_CASTLE_FOREST_TRIAL, world, [
+        (Locations.GANONS_CASTLE_FOREST_TRIAL_CHEST, lambda bundle: can_kill_enemy(bundle, Enemies.WOLFOS)),
+        (Locations.GANONS_CASTLE_FOREST_TRIAL_POT1, lambda bundle: (can_break_pots(bundle) and
+                                                                    (can_use(Items.FIRE_ARROW, bundle) or
+                                                                     (can_use(Items.DINS_FIRE, bundle) or
+                                                                      can_use(Items.HOOKSHOT, bundle))))),
```

Comment:
```suggestion
        (Locations.GANONS_CASTLE_FOREST_TRIAL_POT1, lambda bundle: can_break_pots(bundle) and
                                                                    (can_use(Items.FIRE_ARROW, bundle) or
                                                                     (can_use(Items.DINS_FIRE, bundle) and
                                                                     (can_use_any([Items.FAIRY_BOW,Items.HOOKSHOT], bundle))))),
```
Same with the pot below